### PR TITLE
SI-5943 toolboxes now autoimport Predef and scala

### PR DESCRIPTION
**Pull request by xeno-by**
Previously tb.typeCheck used default typer, which builds upon NoContext.
Changing the context to analyzer.rootContext(NoCompilationUnit, EmptyTree)
fixed the missing imports problem.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @paulp @retronym @gkossakowski

**Comment by xeno-by** [comment]
Duh this one is legit

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by xeno-by** [comment]
Oh come on kitteh

**Comment by xeno-by** [comment]
yeah this failure looks legitimate. will look into it tomorrow

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by xeno-by** [comment]
lgtm anyone?

File: test/pending/run/t5943b1.check [hunk 1]
```diff
@@ -0,0 +1 @@
+2.0
```
Comment (gkossakowski) [inline test/pending/run/t5943b1.check]: This output is wrong right? In the test you are printing the tree:

``` scala
println(tb.typeCheck(expr))
```

Comment (xeno-by) [inline test/pending/run/t5943b1.check]: wow

**Comment by gkossakowski** [comment]
Mention why test is added to pending. I'm guessing because of SI-6393, right?

Mention it both in a comment in a test and commit message.

File: test/pending/run/t5943b1.check (hunk 1 above)
Comment (xeno-by) [inline test/pending/run/t5943b1.check]: I'll remove this check file altogether

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by gkossakowski** [comment]
My understanding of Contexts is very limited but the change looks good to me. Especially because test-cases make it clear that the situation is being improved.

I'll let this be open for a couple of more hours in case @retronym or @paulp want to give a comment.

**Comment by paulp** [comment]
I have no bone to pick with #1331.

**Comment by gkossakowski** [comment]
Cool. Thanks! Merging.

**Pull request merged**